### DISCUSSION TITLE
How can I source my r source code to the root of the project directory?

### DISCUSSION THREAD

**Discussion by jonso99** [quarto-dev/quarto-cli#5213] [Q&A] [answered]
Hello all

I have a quarto R book with a lot of qmd files and would love to keep them in folder named after the book parts, like
```
PartA/intro.qmd
```
Here is what I cannot figure out. Suppose I keep data or a common R file in a folder called data, then I cannot source it with the same path depending on if I render book in terminal or edit qmd it in VScode and run code chunks there. One needs
```
source("common/functions.r")
```
and the other 
```
source("../common/functions.r")
```
So as far as I can see, the only way to make this work is to keep all qmd files in the root folder. Which is very messy. 

One might think that the R `here` library would solve this, but `here` seems incompatible with quarto. 

I must be missing something incredibly stupid.

thanks

jack

**Comment by eitsupi** [comment, marked as answer]
We can use the `execute-dir: project` setting.
https://quarto.org/docs/projects/code-execution.html#working-dir

See also:
https://quarto.org/docs/tools/vscode.html#execution-directory

**Reply by jonso99** [reply]
Thanks a lot @eitsupi 

I knew I was missing something obvious!

best, jack
